src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,125 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN_VALUE = "ABC1234";
+    private static final String VALID_PASSWORD_VALUE = "abc1234";
+    private static final Integer VALID_AGE_VALUE = 18;
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullLoginValue_throwsException() {
+        User user = createUser(null, VALID_PASSWORD_VALUE, VALID_AGE_VALUE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortLoginValue_throwsException() {
+        User user = createUser("abc", VALID_PASSWORD_VALUE, VALID_AGE_VALUE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_edgeLoginValue_throwsException() {
+        User user = createUser("abc12", VALID_PASSWORD_VALUE, VALID_AGE_VALUE);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_validLoginValue_success() {
```

Comment:
one valid case is enough 

```suggestion
    void register_validUser_success() {
```